dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/InstallUtil.java

```diff
@@ -309,6 +328,55 @@ public static String getInstallerExecutable() throws IOException {
 		}
 	}
 	
+	protected static void copyExecutable(Map<OperatingSystem, String> fileMap) throws IOException {
```

Comment:
Is this method largely a copy `getInstallerExecutable`?

I wonder if it might be less duplicate code if we made `installMap` a map of string arrays, then we can just do the copy in a loop.

Reply:
I had wanted to create another method that could be reusable, but there are a few things that are different. First the get installer executable will assign the installExec string to it and the log messages are also pretty specific to it. The file check at the start for getInstallerExecutable is also not done. Originally, I really did want to refactor everything to make it generic, however, there is a higher chance to introduce a new regression and this code will not be used once we switch to downloading the binary instead of including it

Reply:
Could we at least refactor it so the actual part doing the copy (i.e. the try block is in a common private method - minus the `installExec` assignment which can be moved to the outside)